### PR TITLE
Add Windows support on v2 master

### PR DESCRIPTION
**Pull request by kaiguo**
Summary:
---------

Adding Windows async-storage.

Test Plan:
----------

0. [System requirments](https://microsoft.github.io/react-native-windows/docs/rnw-dependencies).
1. Restore modules
```
cd packages\storage-legacy
yarn install
cd ..\..\examples\mobile
yarn install
```
2. Start Metro
```
yarn start:windows
```
3. Open `examples\mobile\AsyncStorageExample.sln` with VisualStudio and run.

Screenshot_20200403113108

### CODE REVIEW

**Comment by Krizzu** [comment]
I wish we could talk this through before any work was done 🙏 
I'm putting v2 work on halt due to limited time constraint and redirecting my focus on "legacy" version.

Can we address Windows platform support there?

**Comment by kaiguo** [comment]
> I wish we could talk this through before any work was done 🙏
> I'm putting v2 work on halt due to limited time constraint and redirecting my focus on "legacy" version.
> 
> Can we address Windows platform support there?

OK, I will open another PR against LEGACY. 

Should we check in this current PR as well? So we'll have Windows support here too when you do start working on v2.

**Comment by kaiguo** [comment]
@Krizzu created another PR here: https://github.com/react-native-community/async-storage/pull/327.

File: packages/storage-legacy/windows/ReactNativeAsyncStorage/RNCAsyncStorage.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "pch.h"
+#include "NativeModules.h"
+#include "DBStorage.h"
+
+using namespace winrt::Microsoft::ReactNative;
```
Comment (NickGerleman) [inline packages/storage-legacy/windows/ReactNativeAsyncStorage/RNCAsyncStorage.h:7]: We shouldn't use "using namespace" in header files as this leaks to anything that includes it.

**Comment by chrisglein** [comment]
@kaiguo with [this pr](https://github.com/react-native-community/async-storage/pull/327) accepted does this one get closed?

**Comment by kaiguo** [comment]
> @kaiguo with [this pr](https://github.com/react-native-community/async-storage/pull/327) accepted does this one get closed?

Yeah closing.

**Pull request closed without merging**